commit c67cba1be192c6f4c7373eef6d43e7cc7d7673d1
Author: jbrunet <jbrunet@baylibre.com>
Date:   Mon Dec 19 16:05:37 2016 +0100

    net: phy: use boolean dt properties for eee broken modes
    
    
    [ Upstream commit 57f3986231bb2c69a55ccab1d2b30a00818027ac ]
    
    The patches regarding eee-broken-modes was merged before all people
    involved could find an agreement on the best way to move forward.
    
    While we agreed on having a DT property to mark particular modes as broken,
    the value used for eee-broken-modes mapped the phy register in very direct
    way. Because of this, the concern is that it could be used to implement
    configuration policies instead of describing a broken HW.
    
    In the end, having a boolean property for each mode seems to be preferred
    over one bit field value mapping the register (too) directly.
    
    Cc: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: Jerome Brunet <jbrunet@baylibre.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/phy/phy_device.c b/drivers/net/phy/phy_device.c
index d9ec748..32b555a 100644
--- a/drivers/net/phy/phy_device.c
+++ b/drivers/net/phy/phy_device.c
@@ -1633,7 +1633,7 @@ static void of_set_phy_supported(struct phy_device *phydev)
 static void of_set_phy_eee_broken(struct phy_device *phydev)
 {
 	struct device_node *node = phydev->mdio.dev.of_node;
-	u32 broken;
+	u32 broken = 0;
 
 	if (!IS_ENABLED(CONFIG_OF_MDIO))
 		return;
@@ -1641,8 +1641,20 @@ static void of_set_phy_eee_broken(struct phy_device *phydev)
 	if (!node)
 		return;
 
-	if (!of_property_read_u32(node, "eee-broken-modes", &broken))
-		phydev->eee_broken_modes = broken;
+	if (of_property_read_bool(node, "eee-broken-100tx"))
+		broken |= MDIO_EEE_100TX;
+	if (of_property_read_bool(node, "eee-broken-1000t"))
+		broken |= MDIO_EEE_1000T;
+	if (of_property_read_bool(node, "eee-broken-10gt"))
+		broken |= MDIO_EEE_10GT;
+	if (of_property_read_bool(node, "eee-broken-1000kx"))
+		broken |= MDIO_EEE_1000KX;
+	if (of_property_read_bool(node, "eee-broken-10gkx4"))
+		broken |= MDIO_EEE_10GKX4;
+	if (of_property_read_bool(node, "eee-broken-10gkr"))
+		broken |= MDIO_EEE_10GKR;
+
+	phydev->eee_broken_modes = broken;
 }
 
 /**
